packages/job-worker/src/blueprints/context/adlibActions.ts

```diff
@@ -636,4 +638,21 @@ export class ActionExecutionContext
 	async hackGetMediaObjectDuration(mediaId: string): Promise<number | undefined> {
 		return getMediaObjectDuration(this._context, mediaId, this.studioId)
 	}
+
+	async executeTSRAction(
+		deviceId: string,
+		actionId: string,
+		payload: Record<string, any>
+	): Promise<TSR.ActionExecutionResult> {
+		const device = await this._context.directCollections.PeripheralDevices.findOne({
+			_id: new RegExp(`${deviceId}$`),
+		})
+		if (!device) {
```

Comment:
As you are only intesrested in the `_id` of the of the device, it would be good to use a `projection` on the query, to avoid other properties (such as the large configmanifest) being parsed and loaded into memory